libraries/AP_ExternalAHRS/AP_ExternalAHRS_InertialLabs.cpp

```diff
@@ -254,165 +254,205 @@ bool AP_ExternalAHRS_InertialLabs::check_uart()
-            CHECK_SIZE(u.position);
-            gps_data.latitude = u.position.lat;
-            gps_data.longitude = u.position.lon;
-            gps_data.msl_altitude = u.position.alt;
-            break;
-        }
-        case MessageType::GNSS_VEL_TRACK: {
-            CHECK_SIZE(u.gnss_vel_track);
-            Vector2f velxy;
-            velxy.offset_bearing(u.gnss_vel_track.track_over_ground*0.01, u.gnss_vel_track.hor_speed*0.01);
-            gps_data.ned_vel_north = velxy.x;
-            gps_data.ned_vel_east = velxy.y;
-            gps_data.ned_vel_down = -u.gnss_vel_track.ver_speed*0.01;
-            break;
-        }
-        case MessageType::GNSS_POS_TIMESTAMP: {
-            CHECK_SIZE(u.gnss_pos_timestamp);
-            gps_data.ms_tow = u.gnss_pos_timestamp;
-            break;
-        }
-        case MessageType::GNSS_INFO_SHORT: {
-            CHECK_SIZE(u.gnss_info_short);
-            state2.gnss_info_short = u.gnss_info_short;
-            break;
-        }
-        case MessageType::GNSS_NEW_DATA: {
-            CHECK_SIZE(u.gnss_new_data);
-            state2.gnss_new_data = u.gnss_new_data;
-            break;
-        }
-        case MessageType::GNSS_JAM_STATUS: {
-            CHECK_SIZE(u.gnss_jam_status);
-            state2.gnss_jam_status = u.gnss_jam_status;
-            break;
-        }
-        case MessageType::DIFFERENTIAL_PRESSURE: {
-            CHECK_SIZE(u.differential_pressure);
-            airspeed_data.differential_pressure = u.differential_pressure*1.0e-4;
-            break;
-        }
-        case MessageType::TRUE_AIRSPEED: {
-            CHECK_SIZE(u.true_airspeed);
-            state2.true_airspeed = u.true_airspeed*0.01;
-            break;
-        }
-        case MessageType::WIND_SPEED: {
-            CHECK_SIZE(u.wind_speed);
-            state2.wind_speed = u.wind_speed.tofloat().rfu_to_frd()*0.01;
-            break;
-        }
-        case MessageType::AIR_DATA_STATUS: {
-            CHECK_SIZE(u.air_data_status);
-            state2.air_data_status = u.air_data_status;
-            break;
-        }
-        case MessageType::SUPPLY_VOLTAGE: {
-            CHECK_SIZE(u.supply_voltage);
-            state2.supply_voltage = u.supply_voltage*0.01;
-            break;
-        }
-        case MessageType::TEMPERATURE: {
-            CHECK_SIZE(u.temperature);
-            // assume same temperature for baro and airspeed
-            baro_data.temperature = u.temperature*0.1;
-            airspeed_data.temperature = u.temperature*0.1;
-            break;
-        }
-        case MessageType::UNIT_STATUS2: {
-            CHECK_SIZE(u.unit_status2);
-            state2.unit_status2 = u.unit_status2;
-            break;
-        }
+            case MessageType::GPS_INS_TIME_MS: {
+                // this is the GPS tow timestamp in ms for when the IMU data was sampled
+                CHECK_SIZE(u.gnss_time_ms);
+                nav_ins_data.ms_tow = u.gnss_time_ms;
```

Comment:
Sorry, you mean this shift about switch-case or something else?
I returned it back to avoid this big diff

Reply:
> Sorry, you mean this shift about switch-case or something else? I returned it back to avoid this big diff

Yes, minimising the diff is important.  It makes review much faster, and if a PR needs reviewing multiple times it can be very frustrating.

In this case you appear to have changed 'gps' to 'gnss' - and while I understand that's a correct change, it's also unnecessary as the two are often used interchangably.

Reply:
Oh, now I understood, sorry. I renamed this because other structures related to GNSS were added to the processor and to the simulator.
And in order to keep all the names consistent with the documentation, I decided to use GNSS everywhere. To make it easier to find in documentation in the future.

But if this diff interferes, then I can rename all gnss (including new ones) to gps.
It will make the finding of these values ​​in the documentation more difficult

Reply:
Please remove the renames from gps to gnss.  New stuff in this PR can use gnss, for now we treat them synonymously in ArduPilot.

This and the whitespace change significantly reduced my enthusiasm to review this PR!

If you want to come back and do a separate PR just to rename things in this new driver for consistency, I think I'd merge it.  The code in here is very young, and such a change is provably correct by ensuring compiler output is identical before/after.

Don't bother reverting the simulator changes - I'm happy enough merging those.

Reply:
Done, I removed all renaming changes